### PR TITLE
Issue #77, Issue #71, fixes to schemas, external references

### PR DESCRIPTION
**Pull request by grechaw**
Still to go with version comparison -- refs, complete test coverage

We may want to review this PR before merging.  I'll go through and triage/mark issues that I think are covered here, and will try to verify. 

In the meantime, take a look -- there are changes to how serialization and extraction works for external references -- something that QA has looked for for a long time.

Version comparison is mostly done in the PR.  If we decide not to merge for now I may add to this branch to complete that featuure

### CODE REVIEW

**Comment by grechaw** [comment]
This is a big PR that completes (I think) version-comparison-generate.  Tests all pass -- there are a lot of updates to keys around external references though, as noted above, I recorded some bugs here and have solutions included.

**Comment by grechaw** [comment]
I'll merge when you're ready for this one.

File: entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy [hunk 1]
```diff
@@ -36,7 +36,7 @@ declare private function es-codegen:casting-function-name(
 {
     if ($datatype eq "iri")
     then "sem:iri"
-    else "xs:" || $datatype
+    else "xs:" || $datatype || "#1"
```
Comment (bsrikan) [inline entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy]: What does "#1" represent

Comment (grechaw) [inline entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy]: In XQuery, one refers to a function literal with the name of the function, and the number of arguments appended.

So map:map#3  is one function.  A casting function has one arg, hence #1

**Comment by bsrikan** [comment]
Thanks. Looks good to me.

**Comment by rashiatmarklogic** [comment]
Looks good, please merge. Also, would be nice if in this week, we can sync up on all the expected changes...

**Pull request merged**